scapy/contrib/mount.py

```diff
@@ -0,0 +1,143 @@
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more information
+# Copyright (C) Lucas Preston <lucas.preston@infinite.io>
+# This program is published under a GPLv2 license
+
+# scapy.contrib.description = NFS Mount v3
+# scapy.contrib.status = loads
+
+from binascii import unhexlify
+from scapy.contrib.oncrpc import RPC, RPC_Call
+from scapy.packet import Packet, bind_layers
+from scapy.fields import IntField, StrLenField, IntEnumField, PacketField, \
+    ConditionalField, FieldListField
+
+mountstat3 = {
+    0: 'MNT3_OK',
+    1: 'MNT3ERR_PERM',
+    2: 'MNT3ERR_NOENT',
+    5: 'MNT3ERR_IO',
+    13: 'MNT3ERR_ACCES',
+    20: 'MNT3ERR_NOTDIR',
+    22: 'MNT3ERR_INVAL',
+    63: 'MNT3ERR_NAMETOOLONG',
+    10004: 'MNT3ERR_NOTSUPP',
+    10006: 'MNT3ERR_SERVERFAULT'
+}
+
+
+class Path(Packet):
+    name = 'Path'
+    fields_desc = [
+        IntField('length', 0),
+        StrLenField('path', '', length_from=lambda pkt: pkt.length),
+        StrLenField('fill', '', length_from=lambda pkt: (4 - pkt.length) % 4)
+    ]
+
+    def extract_padding(self, s):
+        return '', s
+
+    def set(self, path, length=None, fill=None):
+        if length is None:
+            length = len(path)
+        if fill is None:
+            fill = b'\x00' * ((4 - len(path)) % 4)
+        self.length = length
+        self.path = path
+        self.fill = fill
+
+
+class File_Object(Packet):
+    name = 'File Object'
+    fields_desc = [
+        IntField('length', 0),
+        StrLenField('fh', b'', length_from=lambda pkt: pkt.length),
+        StrLenField('fill', b'', length_from=lambda pkt: (4 - pkt.length) % 4)
+    ]
+
+    def set(self, new_filehandle, length=None, fill=None):
+        # convert filehandle to bytes if it was passed as a string
+        if new_filehandle.isalnum():
+            new_filehandle = unhexlify(new_filehandle)
+
+        if length is None:
+            length = len(new_filehandle)
+        if fill is None:
+            fill = b'\x00' * ((4 - self.length) % 4)
```

Comment:
Are you sure you didn't mean `length` rather than `self.length` ? It's only updated later on.
You should have a look at the codacy report:
https://app.codacy.com/app/secdev/scapy/pullRequest?prid=4008484
it detected that

Reply:
`48dc21e4`